Tasks/PublishCodeCoverageResultsV1/task.json

```diff
@@ -43,13 +43,31 @@
             "required": true,
             "helpMarkDown": "Path of the summary file containing code coverage statistics, such as line, method, and class coverage. The value may contain minimatch patterns. For example: `$(System.DefaultWorkingDirectory)/MyApp/**/site/cobertura/coverage.xml`"
         },
+        {
+            "name": "autogenerateHtmlReport",
+            "type": "boolean",
+            "label": "Generate Html report",
+            "defaultValue": "false",
+            "required": false,
+            "helpMarkDown": "Generate code coverage html report."
+        },
+        {
+            "name": "sourceDirectory",
+            "type": "string",
+            "label": "Source directory",
+            "defaultValue": "$(System.DefaultWorkingDirectory)",
+            "required": false,
+            "helpMarkDown": "Path to source code for generating code coverage Html report. For example: `$(System.DefaultWorkingDirectory)/MyApp/`",
+            "visibleRule": "autogenerateHtmlReport = true"
+        },
         {
             "name": "reportDirectory",
             "type": "string",
             "label": "Report directory",
             "defaultValue": "",
             "required": false,
-            "helpMarkDown": "Path of the code coverage HTML report directory. The report directory is published for later viewing as an artifact of the build. The value may contain minimatch patterns. For example: `$(System.DefaultWorkingDirectory)/MyApp/**/site/cobertura`"
+            "helpMarkDown": "Path of the code coverage HTML report directory. The report directory is published for later viewing as an artifact of the build. The value may contain minimatch patterns. For example: `$(System.DefaultWorkingDirectory)/MyApp/**/site/cobertura`",
+            "visibleRule": "false = true"
```

Comment:
I haven't seen this visibleRule before. Doesn't this mean reportDirectory will never be visible? That would be a breaking change in the task.